src/ruby_supportlib/phusion_passenger/vendor/daemon_controller.rb

```diff
@@ -578,45 +551,88 @@ def log_file_has_changed?
 
   def differences_in_log_file
     if @original_log_file_stat && @original_log_file_stat.file?
-      File.open(@log_file, 'r') do |f|
+      File.open(@log_file, "r") do |f|
         f.seek(@original_log_file_stat.size, IO::SEEK_SET)
-        diff = f.read.strip
-        if diff.empty?
-          nil
-        else
-          diff
-        end
+        f.read.strip
       end
-    else
-      nil
     end
   rescue Errno::ENOENT, Errno::ESPIPE
     # ESPIPE means the log file is a pipe.
     nil
   end
 
-  def determine_lock_file(options, identifier, pid_file)
-    if options[:lock_file]
-      LockFile.new(File.expand_path(options[:lock_file]))
+  def determine_lock_file(given_lock_file, identifier, pid_file)
+    if given_lock_file
+      LockFile.new(File.absolute_path(given_lock_file))
     else
-      LockFile.new(File.expand_path(pid_file + ".lock"))
+      LockFile.new(File.absolute_path(pid_file + ".lock"))
     end
   end
 
-  def self.fork_supported?
-    RUBY_PLATFORM != "java" && RUBY_PLATFORM !~ /win32/
-  end
-
-  def self.spawn_supported?
-    # Process.spawn doesn't work very well in JRuby.
-    Process.respond_to?(:spawn) && RUBY_PLATFORM != "java"
-  end
-
   def run_command(command)
     if should_capture_output_while_running_command?
-      run_command_while_capturing_output(command)
+      # Create tempfile for storing the command's output.
+      tempfile = Tempfile.new("daemon-output")
```

Comment:
Cold we use `Tempfile.create` instead? https://rubyapi.org/3.4/o/tempfile#method-c-create